tests/suites/deploy_caas/task.sh

```diff
@@ -11,6 +11,15 @@ test_deploy_caas() {
 
 	file="${TEST_DIR}/test-deploy-caas.log"
 
+	case "${BOOTSTRAP_PROVIDER:-}" in
+  "k8s")
+    test_deploy_aks
+    ;;
+  *)
+    echo "==> TEST SKIPPED: test_deploy_caas test runs on k8s only"
+    ;;
+  esac
```

Comment:
I don't really think this integration test belongs here

We are always running a test in azure if the provider is 'k8s'. Ideally we would treat aks as it's own provider. This is particularly nasty given we bootstrap a controller inside of the unit test itself.

Creating a new aks provider will become necessary if we want any more integration tests, but it'll be a fairly substantial bit of work. My preference in the short term would be to extract this out into its own integration test. It's not nice that we have to bootstrap a controller in aks, successfully run the integration test, then tear down before we even start `test_desploy_charm` in it's own controller. Ideally each integration test suite would only require bootstrapping a single controller